src/main/scala/it/pagopa/pdnd/interop/uservice/partymanagement/api/impl/PartyApiServiceImpl.scala

```diff
@@ -284,13 +287,18 @@ class PartyApiServiceImpl(
       r        <- relationshipsFromParams(fromUuid, toUuid)
     } yield r
 
+    val productsArray     = parseArrayParameter(products)
+    val productRolesArray = parseArrayParameter(productRoles)
+    val rolesArray        = parseArrayParameter(roles)
+    val statesArray       = parseArrayParameter(states)
+
     val filteredResult: Future[List[Relationship]] =
       result.map(relationships =>
         relationships
-          .filter(relationship => product.forall(filter => filter == relationship.product.id))
-          .filter(relationship => productRole.forall(filter => filter == relationship.product.role))
-          .filter(relationship => role.forall(filter => filter == relationship.role.toString))
-          .filter(relationship => state.forall(filter => filter == relationship.state.toString))
+          .filter(relationship => productsArray.isEmpty || productsArray.contains(relationship.product.id))
```

Comment:
do you have any nicer idea?

Reply:
Same problem in party-process.
I use the same approach but through a [Value Class](https://github.com/pagopa/pdnd-interop-uservice-party-process/blob/0.1.x/src/main/scala/it/pagopa/pdnd/interop/uservice/partyprocess/api/impl/package.scala).

Reply:
ok, I'll leave it like this then.

One more point: I was thinking that we could move this filter to the behavior, in order to distribute the execution across shards.
It will make commands a bit more verbose, but it could be worth it for performances (I have no idea of how many results we can expect).

Thoughts?

Reply:
Yes, I had the same idea, but I prefer to leave commands how they are for the same reason (verbosity).
Since we had the same idea, it's probably an acceptable change; the benefit may be more significant than having less verbose commands.

@P3trur0 what do you think?

Reply:
The only thing coming to my mind is something the following:

```scala 
def parseArrayParametersCondition(param: String): String => Boolean = { s =>
   val arrayResult = if (param == "[]") List.empty else param.parseCommaSeparated
    arrayResult.isEmpty || arrayResult.contains(s)
  }
  
  ...
  
  val byProductsCondition     = parseArrayParametersCondition(products)
  val byProductRolesCondition = parseArrayParametersCondition(productRoles)
  val byRolesCondition        = parseArrayParametersCondition(roles)
  val byStatesCondition      = parseArrayParametersCondition(states)

  List("a", "b")
    .filter(byProductsCondition)
    .filter(byProductRolesCondition)
    .filter(byRolesCondition)
    .filter(byStatesCondition)
```

Reply:
see https://github.com/pagopa/pdnd-interop-uservice-party-management/pull/35#discussion_r760282631

Reply:
@P3trur0 what do you think about moving the filter to the behavior?

Reply:
ok for me.